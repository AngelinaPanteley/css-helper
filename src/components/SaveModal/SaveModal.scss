@import '../../variables';

.Modal {
  @include xy-center;

  position: absolute;
  width: auto;
  max-width: 70%;
  height: auto;
  z-index: map-get($z-index, popup);
  background-color: $color-gray;
  box-sizing: border-box;
  transition: transform 0.3s ease-out;
  padding: 40px;
  border-radius: 5px;
  top: 30vh;

  &.Open {
    transform: translate(-50%, 0);
  }

  &.Close {
    transform: translate(-50%, calc(-100% - 30vh));
  }
}

.Form {
  @include flex-wrapper(column, flex-start, flex-start);

  input[type='text'] {
    height: 40px;
    margin-bottom: 40px;
    font-size: 20px;
    padding: 0 10px;
    color: $color-gray;
    border: none;
    outline: none;
    width: 300px;
  }

  label {
    color: white;
    margin-bottom: 20px;
    font-size: 20px;
  }
}

.ButtonContainer {
  @include flex-wrapper(row, flex-end);

  width: 100%;

  * {
    margin-left: 10px;
  }
  
  input, 
  button {
    text-decoration: none;
    color: $color-white;
    transition: $transition;
    font-size: 15px;
    text-transform: uppercase;
    padding: 9px 30px 10px;
    border: 1px solid $color-white;
    background: rgba(56, 58, 71, 0.55);
    align-self: flex-end;
    cursor: pointer;
    outline: none;
  }

  button{
    &:hover,
    &:focus,
    &:active {
      background: rgba(255, 255, 255, 0.1);
      transition: $transition;
      color: $color-white;
    }
  }

  input {
    background: rgba(255, 0, 81, 0.4);

    &[disabled] {
      cursor: not-allowed;
    }

    &:hover,
    &:focus[disabled=false],
    &:active[disabled=false] {
      &:not([disabled]) {
        background: rgba(255, 0, 81, 0.6);
        transition: $transition;
        color: $color-white;
      }
    }
  }
}