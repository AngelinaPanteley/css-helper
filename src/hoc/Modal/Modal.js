import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styles from './Modal.scss';
import Backdrop from '../../components/UI/Backdrop/Backdrop';
import Auxiliary from '../Auxiliary/Auxiliary';

class Modal extends PureComponent {
  static propTypes = {
    isOpen: PropTypes.bool.isRequired,
    onClose: PropTypes.func.isRequired,
  }

  render() {
    const attachedClasses = [styles.Modal];
    if (this.props.isOpen) {
      attachedClasses.push(styles.Open);
    } else {
      attachedClasses.push(styles.Close);
    }
    return (
      <Auxiliary>
        <Backdrop
          show={this.props.isOpen}
          clicked={this.props.onClose}
          withBackground={true} />
        <div className={attachedClasses.join(' ')}>
          {this.props.children}
        </div>
      </Auxiliary>
    );
  }
}

export default Modal;