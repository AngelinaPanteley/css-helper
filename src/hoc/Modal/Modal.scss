@import '../../variables';

.Modal {
  @include xy-center;

  position: absolute;
  width: auto;
  max-width: 70%;
  height: auto;
  z-index: map-get($z-index, popup);
  background-color: $color-gray;
  box-sizing: border-box;
  transition: transform 0.3s ease-out;
  padding: 40px;
  border-radius: 5px;
  top: 30vh;

  &.Open {
    transform: translate(-50%, 0);
  }

  &.Close {
    transform: translate(-50%, calc(-100% - 30vh));
  }
}