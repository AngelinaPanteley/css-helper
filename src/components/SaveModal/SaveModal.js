import React, { PureComponent } from 'react';
import styles from './SaveModal.scss';
import Modal from '../../hoc/Modal/Modal';

class SaveModal extends PureComponent {
  state = {
    title: '',
  }

  handleChange = (e) => {
    this.setState({
      [e.target.name]: e.target.value,
    });
  }

  handleSubmit = (e) => {
    this.props.onClose(e);
    this.props.onSubmit(this.state.title);
  }

  render() {
    return (
      <Modal isOpen={this.props.isOpen} onClose={this.props.onClose}>
        <form onSubmit={this.handleSubmit} className={styles.Form}>
          <label htmlFor='title'>Enter title:</label>
          <input id='title' name='title' type="text" onChange={this.handleChange} />
          <div className={styles.ButtonContainer}>
            <button onClick={this.props.onClose}>Cancel</button>
            <input type="submit" value="Save" disabled={!this.state.title} />
          </div>
        </form>
      </Modal>
    );
  }
}

export default SaveModal;