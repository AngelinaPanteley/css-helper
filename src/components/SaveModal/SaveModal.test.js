"use strict";

import React from 'react';
import renderer from 'react-test-renderer';
import { expect as chaiExpect } from 'chai';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16.1';
import Modal from '../../hoc/Modal/Modal';
import SaveModal from './SaveModal.js';

configure({ adapter: new Adapter() });

describe('>>>SaveModal', () => {

  //SNAPSHOTS

  const props = {
    isOpen: true,
    onClose: () => { },
  };

  const modalComponent = <SaveModal {...props} />;

  const component = renderer.create(modalComponent);

  it('renders save modal correctly', () => {
    let componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();
  });

  //SHALLOW RENDERING

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(modalComponent);
  });

  it('renders modal hoc', () => {
    chaiExpect(wrapper.find(Modal)).to.have.lengthOf(1);
  });

  it('renders one form', () => {
    chaiExpect(wrapper.find('form')).to.have.lengthOf(1);
  });
})