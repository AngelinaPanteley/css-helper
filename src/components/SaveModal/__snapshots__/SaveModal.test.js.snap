// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>SaveModal renders save modal correctly 1`] = `
Array [
  <div
    className=" "
    onClick={[Function]}
  />,
  <div
    className=" "
  >
    <form
      onSubmit={[Function]}
    >
      <label
        htmlFor="title"
      >
        Enter title:
      </label>
      <input
        id="title"
        name="title"
        onChange={[Function]}
        type="text"
      />
      <div>
        <button
          onClick={[Function]}
        >
          Cancel
        </button>
        <input
          disabled={true}
          type="submit"
          value="Save"
        />
      </div>
    </form>
  </div>,
]
`;
